File: php/TwoSum.php
<?php
class Solution {
    /**
     * @param Integer[]
     * @param Integer
     * @return Integer[]
     */
    function twoSum($nums, $target) {
        $hasher = [];
        foreach ($nums as $idx => $num) {
            $complement = $target - $num;
            if (array_key_exists($complement, $hasher)) {
                return [$hasher[$complement], $idx];
            }
            $hasher[$num] = $idx;
        }
        return [];
    }
}

$nums = [2, 7, 11, 15];
$target = 9;

$sol = new Solution();
$result = $sol->twoSum($nums, $target);
print_r($result);

File: Python/TwoSum.py
from typing import List


class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        hasher = {}
        for idx, num in enumerate(nums):
            complement = target - num
            if complement in hasher:
                return [hasher[complement], idx]
            hasher[num] = idx


nums = [2, 7, 11, 15]
target = 9

sol = Solution()
resultado = sol.twoSum(nums, target)
print(resultado)